packages/accounts-controller/src/AccountsController.test.ts

```diff
@@ -0,0 +1,1203 @@
+import { ControllerMessenger } from '@metamask/base-controller';
+import type { InternalAccount } from '@metamask/keyring-api';
+import type { KeyringControllerState } from '@metamask/keyring-controller';
+import type { SnapControllerState } from '@metamask/snaps-controllers';
+import { SnapStatus } from '@metamask/snaps-utils';
+import * as uuid from 'uuid';
+
+import type {
+  AccountsControllerActions,
+  AccountsControllerEvents,
+  AccountsControllerState,
+} from './AccountsController';
+import { AccountsController, keyringTypeToName } from './AccountsController';
+
+jest.mock('uuid');
+const mockUUID = jest.spyOn(uuid, 'v4');
+
+const defaultState: AccountsControllerState = {
+  internalAccounts: {
+    accounts: {},
+    selectedAccount: '',
+  },
+};
+
+const mockGetKeyringForAccount = jest.fn();
+const mockGetKeyringByType = jest.fn();
+const mockGetAccounts = jest.fn();
+
+const mockAccount: InternalAccount = {
+  id: 'mock-id',
+  address: '0x123',
+  options: {},
+  methods: [
```

Comment:
We added a `EthMethod` enum in the `keyring-api`, it might be a good idea to use it here.